src/main/java/com/thebois/controllers/toolbar/PlaceStructureController.java

```diff
@@ -0,0 +1,84 @@
+package com.thebois.controllers.toolbar;
+
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+
+import com.thebois.Pawntastic;
+import com.thebois.controllers.AbstractInputProcessor;
+import com.thebois.controllers.IController;
+import com.thebois.listeners.IEventListener;
+import com.thebois.listeners.events.OnClickEvent;
+import com.thebois.models.world.World;
+import com.thebois.models.world.structures.StructureType;
+import com.thebois.views.IProjector;
+import com.thebois.views.info.IActorView;
+import com.thebois.views.toolbar.StructureToolbarView;
+
+/**
+ * Controller that manages the structure buttons and is responsible for placing structures.
+ */
+public class PlaceStructureController extends AbstractInputProcessor
+    implements IEventListener<OnClickEvent<StructureType>>, IController<IActorView> {
+
+    private static final int TILE_SIZE = Pawntastic.getTileSize();
+    private final World world;
+    private final IProjector projector;
+    private Actor toolbarWidget;
+    private StructureType selectedStructure = StructureType.HOUSE;
+    private final StructureToolbarView structureToolbarView;
+
+    /**
+     * Instantiate a controller that manage placement of structures.
+     *
+     * @param world     The world where the structures should be placed.
+     * @param projector Projector used to translate screen coordinates to world coordinates.
+     * @param skin      The skin used to create the buttons.
+     */
+    public PlaceStructureController(
+        final World world, final IProjector projector, final Skin skin) {
+        this.world = world;
+        this.projector = projector;
+
+        structureToolbarView = new StructureToolbarView(skin, this);
+    }
+
+    @Override
+    public boolean touchDown(final int x, final int y, final int pointer, final int button) {
+        if (button == LEFT_CLICK) {
+            final Vector2 worldSpaceCoordinates = projector.unproject(x, y);
+            final Actor gameContainer = toolbarWidget.getParent();
+            final float offSetX = gameContainer.getX();
+            final float offSetY = gameContainer.getY() + toolbarWidget.getHeight();
+            final int worldPosX = (int) ((worldSpaceCoordinates.x - offSetX) / TILE_SIZE);
+            final int worldPosY = (int) ((worldSpaceCoordinates.y - offSetY) / TILE_SIZE);
+            return world.createStructure(selectedStructure, worldPosX, worldPosY);
+        }
+        return false;
+    }
+
+    /**
+     * Set the game widget used to get the offset.
+     *
+     * @param toolbarWidget The widget in which the game is rendered.
+     */
+    public void setToolbarWidget(final Actor toolbarWidget) {
+        this.toolbarWidget = toolbarWidget;
+    }
```

Comment:
Why not just set the offset?

Reply:
When this method is being called, the widget itself doesn't have a height or width. I couldn't figure out why it didn't, but I think it's because it has not been rendered yet which it does in the first render loop.

Reply:
Oh, well that's unfortunate. Can't you inject the toolbar in the constructor instead, then?

Reply:
Nope because the toolbar is created after the controllers.

Reply:
Welp, guess it'll have to stay like this then. Still feels a bit wrong, but at least it works!